glib2, libfido2: remove PKG_FORTIFY_SOURCE workarounds

ynezz opened the pull request:
Maintainer: @neheb @stintel @linosgian
Compile tested: snapshot, aarch64_cortex-a53_gcc-12.3.0_musl, PKG_FORTIFY_SOURCE_2=y
Run tested: N/A

Description:

While reviewing https://github.com/openwrt/openwrt/pull/13001 I've tried to reproduce the mentioned issue with glib2 and libfido2 and found out, that this workaround is probably not needed anymore and should be removed.

ynezz committed:
Revert "glib2: disable fortify source"

This reverts commit 723c4360c48a8483ab059e65dcd4a51cd6a5925b as it
seems, that its not needed anymore, build tested with
aarch64_cortex-a53_gcc-12.3.0_musl and PKG_FORTIFY_SOURCE_2=y

Signed-off-by: Petr Štetiar <ynezz@true.cz>

ynezz committed:
libfido2: drop PKG_FORTIFY_SOURCE=0 workaround

Its not needed anymore, build tested with
aarch64_cortex-a53_gcc-12.3.0_musl and PKG_FORTIFY_SOURCE_2=y.

Signed-off-by: Petr Štetiar <ynezz@true.cz>

neheb commented:
@stintel can still reproduce AFAIK.

ynezz committed:
libfido2: fix compilation with _FORTIFY_SOURCE=1

Library forces _FORTIFY_SOURCE=2 and thus when we configure
_FORTIFY_SOURCE=1 we create following compile failure:

 <command-line>: error: "_FORTIFY_SOURCE" redefined

Signed-off-by: Petr Štetiar <ynezz@true.cz>

stintel commented:
Confirmed, glib2 still fails to build for me on qoriq.
[compile.txt](https://github.com/openwrt/packages/files/11904110/compile.txt)
[qoriq-m300-diffconfig.txt](https://github.com/openwrt/packages/files/11904123/qoriq-m300-diffconfig.txt)

ynezz committed:
glib2: disable FORTIFY_SOURCE on powerpc64

It seems, that powerpc64 is still not able to compile glib2 with
FORTIFY_SOURCE enabled:

 [81/481] Compiling C object glib/libglib-2.0.so.0.7400.0.p/gkeyfile.c.o
 FAILED: glib/libglib-2.0.so.0.7400.0.p/gkeyfile.c.o
 powerpc64-openwrt-linux-musl-gcc -Iglib/libglib-2.0.so.0.7400.0.p -Iglib -I../glib -I. -I.. -Iopenwrt/staging_dir/target-powerpc64_e5500_musl/usr/include -Iopenwrt/staging_dir/toolchain-powerpc64_e5500_gcc-12.3.0_musl/usr/include -Iopenwrt/staging_dir/toolchain-powerpc64_e5500_gcc-12.3.0_musl/include/fortify -Iopenwrt/staging_dir/toolchain-powerpc64_e5500_gcc-12.3.0_musl/include -flto=auto -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu99 -D_GNU_SOURCE -fno-strict-aliasing -DG_DISABLE_CAST_CHECKS -DG_DISABLE_ASSERT -Wduplicated-branches -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-cast-function-type -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -Wstrict-prototypes -Wno-bad-function-cast -Werror=declaration-after-statement -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=pointer-sign -Os -pipe -mcpu=e5500 -g3 -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=openwrt/build_dir/target-powerpc64_e5500_musl/glib-2.74.0=glib-2.74.0 -ffunction-sections -fdata-sections -DPIC -fpic -fstack-protector-strong -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -fPIC -pthread '-DG_LOG_DOMAIN="GLib"' -DGLIB_COMPILATION -fvisibility=hidden -MD -MQ glib/libglib-2.0.so.0.7400.0.p/gkeyfile.c.o -MF glib/libglib-2.0.so.0.7400.0.p/gkeyfile.c.o.d -o glib/libglib-2.0.so.0.7400.0.p/gkeyfile.c.o -c ../glib/gkeyfile.c
 In file included from ../glib/gkeyfile.c:35:
 openwrt/staging_dir/toolchain-powerpc64_e5500_gcc-12.3.0_musl/include/fortify/stdio.h: In function 'snprintf':
 openwrt/staging_dir/toolchain-powerpc64_e5500_gcc-12.3.0_musl/include/fortify/stdio.h:101:9: error: format not a string literal, argument types not checked [-Werror=format-nonliteral]
   101 |         return __orig_snprintf(__s, __n, __f, __builtin_va_arg_pack());
       |         ^~~~~~
 openwrt/staging_dir/toolchain-powerpc64_e5500_gcc-12.3.0_musl/include/fortify/stdio.h: In function 'sprintf':
 openwrt/staging_dir/toolchain-powerpc64_e5500_gcc-12.3.0_musl/include/fortify/stdio.h:110:17: error: format not a string literal, argument types not checked [-Werror=format-nonliteral]
   110 |                 __r = __orig_snprintf(__s, __b, __f, __builtin_va_arg_pack());
       |                 ^~~
 openwrt/staging_dir/toolchain-powerpc64_e5500_gcc-12.3.0_musl/include/fortify/stdio.h:114:17: error: format not a string literal, argument types not checked [-Werror=format-nonliteral]
   114 |                 __r = __orig_sprintf(__s, __f, __builtin_va_arg_pack());
       |                 ^~~
 cc1: some warnings being treated as errors

So lets enable PKG_FORTIFY_SOURCE workaround on the affected platform
only.

References: https://github.com/openwrt/packages/pull/21467#issuecomment-1612694426
Signed-off-by: Petr Štetiar <ynezz@true.cz>

ynezz commented:
> Confirmed, glib2 still fails to build for me on qoriq.

Thanks, looks like some `powerpc64` related issue, should [so lets keep it disabled on that platform only](https://github.com/openwrt/packages/pull/21467/commits/db43e1058020ae5c04602acadb79c22ad3cdffff)? I would love to check what is failing, but its probably too deep rabbit hole and don't have much time right now to fix that properly, sorry.

ksperling-apple commented:
Please see my comment here: https://github.com/openwrt/openwrt/pull/13001#issuecomment-1613870804
